manifests/profile/ceilometer/api.pp

```diff
@@ -45,14 +45,19 @@
     require => Class['mongodb::server'],
   }
 
-  mongodb_user { 'ceilometer':
-    ensure        => present,
-    password_hash => mongodb_password('ceilometer', 'password'),
-    database      => 'ceilometer',
-    roles         => ['readWrite', 'dbAdmin'],
-    tries         => 10,
-    require       => [Class['mongodb::server'], Class['mongodb::client']],
+  $mongo_username = $::openstack::config::ceilometer_mongo_username
+  $mongo_password = $::openstack::config::ceilometer_mongo_password
+
+  if $mongo_username and $mongo_password {
+    mongodb_user { $mongo_username:
+      ensure        => present,
+      password_hash => mongodb_password($mongo_username, $mongo_password),
+      database      => 'ceilometer',
+      roles         => ['readWrite', 'dbAdmin'],
+      tries         => 10,
+      require       => [Class['mongodb::server'], Class['mongodb::client']],
+    }
   }
 
-  Class['::mongodb::server'] -> Class['::mongodb::client'] -> Exec['ceilometer-dbsync']
+  Class['::mongodb::server'] -> Class['::mongodb::client'] -> Mongodb_user[$mongo_username] -> Exec['ceilometer-dbsync']
```

Comment:
Mmm, if the `mongodb_user` resource is not defined then this will fail. Could you add `before => Exec['celiometer-dbsync']` to the `mongodb_user` resource inside the `if`?